lib/screens/detail/nps_answer.dart

```diff
@@ -0,0 +1,131 @@
+import 'package:flutter/material.dart';
+import 'package:survey_flutter/model/survey_answer_model.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
+class NpsAnswer extends StatefulWidget {
+  final ValueChanged<String> onChoiceClick;
+  final List<SurveyAnswerModel> answers;
+
+  const NpsAnswer({
+    required this.onChoiceClick,
+    required this.answers,
+    super.key,
+  });
+
+  @override
+  State<StatefulWidget> createState() => NpsAnswerState();
+}
+
+class NpsAnswerState extends State<NpsAnswer> {
+  TextTheme get _textTheme => Theme.of(context).textTheme;
+
+  AppLocalizations get _localizations => AppLocalizations.of(context)!;
+  int? _selectedIndex;
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      mainAxisAlignment: MainAxisAlignment.center,
+      crossAxisAlignment: CrossAxisAlignment.center,
+      children: [
+        Row(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            for (int index = 0; index < widget.answers.length; index++) ...[
+              _buildNpsItem(
+                answer: widget.answers[index],
+                index: index,
+                isStartItem: index == 0,
+                isEndItem: index == (widget.answers.length - 1),
+              )
+            ]
+          ],
+        ),
+        Padding(
+          padding: const EdgeInsets.symmetric(horizontal: 20, vertical: 16),
+          child: Row(
+            children: [
+              Text(
+                _localizations.surveyNpsNotAtAllLikely,
+                style: _textTheme.labelMedium?.copyWith(color: () {
+                  if (_selectedIndex != null &&
+                      (_selectedIndex ?? 0) <
+                          widget.answers.length / 2.round()) {
+                    return Colors.white;
+                  } else {
+                    return Colors.white.withOpacity(0.5);
+                  }
+                }()),
```

Comment:
@kaungkhantsoe sorry but we don't allow writing an anonymous function directly to get a value like this. It's not friendly and makes us confused when looking at the code.

Normally we have two approaches:
- Extract to a function such as `getNpsTextColor`.
- Apply condition like this:
```suggestion
                style: _textTheme.labelMedium?.copyWith(
                  color: (_selectedIndex != null &&
                          (_selectedIndex ?? 0) <
                              widget.answers.length / 2.round())
                      ? Colors.white
                      : Colors.white.withOpacity(0.5),
                ),
```